cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -152,6 +161,19 @@ public synchronized void create(final NamespaceMeta metadata) throws Exception {
     if (!(Principal.SYSTEM.equals(principal) && NamespaceId.DEFAULT.equals(namespace))) {
       authorizationEnforcer.enforce(instanceId, principal, Action.ADMIN);
       authorizer.grant(namespace, principal, ImmutableSet.of(Action.ALL));
+      // Also grant the namespace user all privileges on the namespace, if kerberos is enabled
+      if (kerberosEnabled) {
+        ImpersonationInfo impersonationInfo = impersonationUserResolver.getImpersonationInfo(metadata);
+        String namespacePrincipal = impersonationInfo.getPrincipal();
+        Preconditions.checkNotNull(
```

Comment:
This should not happen, but this is a guard for the case when kerberos is enabled, impersonation info is not set, and ``Constants.Security.CFG_CDAP_MASTER_KRB_PRINCIPAL`` is also not set in ``cdap-site.xml``.